pkg/endpoint/public_ip_watcher.go

```diff
@@ -63,28 +64,44 @@ func (p *PublicIPWatcher) Run(stopCh <-chan struct{}) {
 }
 
 func (p *PublicIPWatcher) syncPublicIP() {
+	var publicIPs []string
 	localEndpointSpec := p.config.LocalEndpoint.Spec()
 
-	publicIP, resolver, err := getPublicIP(p.config.SubmSpec, p.config.K8sClient, localEndpointSpec.BackendConfig, false)
-	if err != nil {
-		logger.Warningf("Could not determine public IP of the gateway node %q: %v", localEndpointSpec.Hostname, err)
-		return
+	publicIPsChanged := false
+
+	for _, family := range p.config.SubmSpec.GetIPFamilies() {
+		if family == k8snet.IPv4 {
+			publicIP, resolver, err := getPublicIP(p.config.SubmSpec, p.config.K8sClient, localEndpointSpec.BackendConfig, false)
+			if err != nil {
+				logger.Warningf("Could not determine public IPV4 of the gateway node %q: %v", localEndpointSpec.Hostname, err)
+			} else {
+				if publicIP != localEndpointSpec.GetPublicIP(k8snet.IPv4) {
+					publicIPsChanged = true
+
+					logger.Infof("Public IPv4 changed for the Gateway, updating the local endpoint with public IP %q obtained from resolver %q",
+						publicIP, resolver)
+				}
+
+				publicIPs = append(publicIPs, publicIP)
+			}
+		} // TODO_IPV6: add getPublicIPV6
```

Comment:
Again, this is simplified and consistent if we keep the family-specific logic in one place (`getPublicIP`). 

```suggestion
	        publicIP, resolver, err := getPublicIP(family, p.config.SubmSpec, p.config.K8sClient, localEndpointSpec.BackendConfig, false)
		if err != nil {
		    logger.Warningf("Could not determine public IP for family %v of the gateway node %q: %v", family, localEndpointSpec.Hostname, err)
		    continue
		} 
		
		if publicIP != localEndpointSpec.GetPublicIP(family) {
		   publicIPs = append(publicIPs, publicIP)
		   
		   logger.Infof( "Public IP for family %v changed for the Gateway, updating the local endpoint with public IP %q obtained from resolver %q", family, publicIP, resolver)
		}
```